src/app/[locale]/incident/components/IncidentDescriptionForm.tsx

```diff
@@ -33,14 +37,39 @@ export const IncidentDescriptionForm = () => {
   const form = useForm<z.infer<typeof incidentDescriptionFormSchema>>({
     resolver: zodResolver(incidentDescriptionFormSchema),
     defaultValues: {
-      description: signal.text,
+      description: formState.description,
     },
   })
 
+  const { description } = form.watch()
+
+  useEffect(() => {
+    const debouncedWatch = debounce(async (value) => {
+      if (value) {
+        const { main, sub } = await getCategoryForDescription(value)
+
+        updateForm({
+          ...formState,
+          category: main,
+          sub_category: sub,
+          sub_category_url:
+            process.env.NEXT_PUBLIC_BASE_URL_API +
+            `/signals/v1/public/terms/categories/${main}/sub_categories/${sub}`,
```

Comment:
Can we move constructing the URL to the POST action, and just keep the main- and sub slug in the store?